### ISSUE TITLE
Add headerAria prop to Dialog 

### ISSUE THREAD

**Issue by karinamzalez** [CMSgov/design-system#1646]
**Describe the bug**
- Dialog currently only enables the addition of aria-labels to the modal parent component. The Dialog accepts headerClassNames but the header also needs an aria-label for screen reader accessibility. 

**To Reproduce**
1. Go to a CMS Dialog
2. Inspect element
3. Notice header does not have an aria-label

**Expected behavior**
- we should be able to specify aria labels for nested components for better accessibility. 

**Desktop (please complete the following information):**
 - OS:  MacOS
 - Browser Chroms

**Comment by zarahzachz** [comment]
Thanks for reporting this. If you want to monitor its progress, you can set yourself as a watcher on this [Jira ticket](https://jira.cms.gov/browse/WNMGDS-1693).

**Comment by forrestbaer** [comment]
This has been resolved with #2062
